### PR TITLE
Functioning Serialize Deserialize (one could say serde)

### PR DESCRIPTION
**Pull request by Hyrum-Coleman**
Sadly it's not actually serde because that would mean we were in rust :(

Utilizing the arduino json library to do json ser/de. Lets us easily send a JsonDocument type across XBEE using serializeJson() for sending, and deserializeJson() for receiving. They take in the JsonDocument to mutate, and the serial channel to send/read to/from

### CODE REVIEW

**Comment by Hyrum-Coleman** [comment]
Realizing that adding the libraries makes it very hard to diff inspect the .ino files that actually matter.

For reference they're at the very bottom of the list.

File: Code/Robot/input.json
Comment (nawper02) [inline Code/Robot/input.json:1]: This is so insanely cool.

File: Code/Robot/UnoSending/UnoSending.ino
Comment (nawper02) [inline Code/Robot/UnoSending/UnoSending.ino:1]: So I see the input.json, where in UnoSending.ino do we attach that json to our JsonDocument object?

**Review by nawper02 — APPROVED** [review]
Very big. Its gonna be so powerful packing info in jsons instead of trying to write a parser. Idea on how we can structure some of our information:
```
{
  "chassis": {
    "side1": {
      "occupiedSpots": [0, 1, 0],
    },
    "side2": {
      "occupiedSpots": [0, 0, 0],
    },
    "side3": {
      "occupiedSpots": [0, 0, 0],
    },
    "top": [0, 0, 0, 1, 0, 0, 0, 0]
  },
  "moves": [
    {
      "action": "drive forward",
      "duration": 1
    },
    {
      "action": "rotate",
      "degrees": 90
    },
    {
      "action": "drive backward",
      "duration": 1
    },
    {
      "action": "strafe sideways",
      "duration": 0.5
    }
    {
      "action": "platform up",
    },
    {
      "action": "belt forward",
      "duration": 0.5
    },
  ],
  "signal": "start"
}
```

File: Code/Robot/UnoSending/UnoSending.ino
Comment (Hyrum-Coleman) [reply Code/Robot/UnoSending/UnoSending.ino:1]: So right now nothing is being added to the object besides "d": 1. I am working on reading from serial input and having that modify the JsonDocument. I made it to copy paste originally, but we could set it up to read directly from the file.

Comment (nawper02) [reply Code/Robot/UnoSending/UnoSending.ino:1]: Gotcha, makes sense. Having it read from the file, which we can edit on the fly in a text editor, would be very clean indeed

**Pull request merged**